src/Sponsor/SponsorManagement.tsx

```diff
@@ -0,0 +1,220 @@
+            contractURL: sponsorDetails.contractURL,
+            nominees: sponsorDetails.nominees
+          }}
+          onSubmit={this.handleSubmit}
+          render={this.renderFormik}
+          validationSchema={getValidationSchema()}
+        />
+      </MaxWidthBox>
+    )
+  }
+
+  public async componentDidMount() {
+    const { mode } = this.state;
+    if (mode === ManageSponsorModes.EDIT) {
+      try {
+        const response = await Sponsor.getSelf();
+        const sponsorDetails = response.data.data;
+        this.setState({ sponsorDetails });
+      } catch (e) {
+        if (e && e.data) {
+          ValidationErrorGenerator(e.data);
+        }
+        // For some reason we could not get self. We should switch our state to CREATE.
+        this.setState({ mode: ManageSponsorModes.CREATE });
+      }
+    }
+  }
+
+  private renderFormik(fp: FormikProps<any>) {
+    return (
+      <Form onSubmit={fp.handleSubmit}>
+        <FastField
+          id="companyName"
+          name={'company'}
+          component={FormikElements.Input}
+          label={CONSTANTS.SPONSOR_COMPANY_LABEL}
+          required={true}
+        />
+        <ErrorMessage
+          component={FormikElements.Error}
+          name="company"
+        />
+
+        <FastField
+          name={'contractURL'}
+          inputType="url"
+          component={FormikElements.Input}
+          label={CONSTANTS.SPONSOR_CONTRACT_URL_LABEL}
+          required={true}
+        />
+        <ErrorMessage
+          component={FormikElements.Error}
+          name="contractURL"
+        />
+
+        <SubmitBtn isLoading={fp.isSubmitting} disabled={fp.isSubmitting}>
+          Submit
+        </SubmitBtn>
+      </Form>
+    )
+  }
+
+  private async handleSubmit(values: any) {
+    try {
+      await this.submit(values);
+      this.setState({ formSubmitted: true });
+    }
+    catch (e) {
+      if (e && e.data) {
+        ValidationErrorGenerator(e.data);
+      }
+    }
+  }
+
+  private async submit(values: any): Promise<void> {
+    const acctResponse = await Account.getSelf();
+
+    const account = acctResponse.data.data;
+    let sponsorTier = 0;
+
+    switch (account.accountType) {
+      case UserType.SPONSOR_T1:
+        sponsorTier = 1;
+        break;
+      case UserType.SPONSOR_T2:
+        sponsorTier = 2;
+        break;
+      case UserType.SPONSOR_T3:
+        sponsorTier = 3;
+        break;
+      case UserType.SPONSOR_T4:
+        sponsorTier = 4;
+        break;
+      case UserType.SPONSOR_T5:
+        sponsorTier = 5;
+        break;
+      default:
+        sponsorTier = -1;
+    }
+
+    const sponsorApplication = this.convertFormikToSponsor(values, account.id, sponsorTier);
+
+    switch (this.state.mode) {
+      case ManageSponsorModes.CREATE:
+        await Sponsor.create(sponsorApplication);
+        break;
+      case ManageSponsorModes.EDIT:
+        const sponsorResponse = await Sponsor.getSelf();
```

Comment:
You should just pass in state.sponsorDetails.id into the convertFormikToSponsor function, and get rid of this call.